app/navigation/Navigator.scala

```diff
@@ -382,38 +389,42 @@ class Navigator @Inject() () {
       .getOrElse(routes.JourneyRecoveryController.onPageLoad())
 
   private def navigateFromAssessmentCheck(assessmentPage: AssessmentPage)(answers: UserAnswers): Call = {
-    val recordId = assessmentPage.recordId
+    if (!assessmentPage.shouldRedirectToCya) {
+      val recordId = assessmentPage.recordId
 
-    for {
-      recordQuery      <- answers.get(RecordCategorisationsQuery)
-      record           <- recordQuery.records.get(recordId)
-      assessmentAnswer <- answers.get(assessmentPage)
-    } yield assessmentAnswer match {
-      case AssessmentAnswer.Exemption(_) =>
-        val assessmentCount = Try {
-          recordQuery.records(recordId).categoryAssessments.size
-        }.getOrElse(0)
-
-        if (assessmentPage.index + 1 < assessmentCount) {
-          if (answers.isDefined(AssessmentPage(recordId, assessmentPage.index + 1))) {
-            routes.CyaCategorisationController.onPageLoad(recordId)
+      for {
+        recordQuery      <- answers.get(RecordCategorisationsQuery)
+        record           <- recordQuery.records.get(recordId)
+        assessmentAnswer <- answers.get(assessmentPage)
+      } yield assessmentAnswer match {
+        case AssessmentAnswer.Exemption(_) =>
+          val assessmentCount = Try {
```

Comment:
do we want to default to 0 here

Reply:
I think it's fine. By which I mean it was already there and is there on the normal mode version too 😄 

Guess it's to avoid having to map the value. Probably should be going to JourneyRecovery if its not defined, but I'm kinda happy to say its existing technical debt and run away?